backends/tfhe-cuda-backend/cuda/src/crypto/keyswitch.cuh

```diff
@@ -76,15 +101,16 @@ __host__ void cuda_keyswitch_lwe_ciphertext_vector(
     uint32_t base_log, uint32_t level_count, uint32_t num_samples) {
 
   cudaSetDevice(gpu_index);
-  constexpr int ideal_threads = 1024;
-  if (lwe_dimension_out + 1 > ideal_threads)
-    PANIC("Cuda error (keyswitch): lwe dimension size out should be greater "
-          "or equal to the number of threads per block")
+
+  constexpr int num_threads_x = 16;
+  constexpr int num_threads_y = 32;
 
   int lwe_size = lwe_dimension_out + 1;
-  int shared_mem = sizeof(Torus) * lwe_size;
-  dim3 grid(num_samples, 1, 1);
-  dim3 threads(ideal_threads, 1, 1);
+  int num_blocks = (lwe_size + num_threads_x - 1) / num_threads_x;
```

Comment:
Let's use `getNumBlocksAndThreads()` in `utils/kernel_dimensions.cuh` to calculate the number of blocks.

Reply:
I saw that function only applies to unidimensional blocks in which each threads produces one output. I created a similar function that works for 2D blocks for which you set the number of threads in y that are going to participate in the calculation of one output.